### PR TITLE
added H2 dialect support

### PR DESCRIPTION
**Pull request by Vendeetta**

### CODE REVIEW

File: entity2jooq-codegen/src/test/java/io/github/devastool/entity2jooq/codegen/type/dialect/H2TypeMapperTest.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests of {@link H2TypeMapper}.
+ *
+ * @author Evgeniy_Gerasimov
+ */
+class H2TypeMapperTest {
+
```
Comment (andrey-yurzanov) [inline entity2jooq-codegen/src/test/java/io/github/devastool/entity2jooq/codegen/type/dialect/H2TypeMapperTest.java:29]: unnecessary line

File: entity2jooq-codegen/src/test/java/io/github/devastool/entity2jooq/codegen/type/dialect/H2TypeMapperTest.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    var dialect = mapper.getDialect();
+    Assertions.assertEquals("H2", dialect);
+  }
+
+  @Test
+  void getSqlTypeTest() {
+    String sqlType = mapper.getSqlType("H2", LocalDateTime.class);
+    Assertions.assertEquals("timestamp", sqlType);
+  }
+
```
Comment (andrey-yurzanov) [inline entity2jooq-codegen/src/test/java/io/github/devastool/entity2jooq/codegen/type/dialect/H2TypeMapperTest.java:43]: unnecessary line

File: entity2jooq-example/entity2jooq-example-h2/src/main/java/io/github/devastool/entity2jooq/example/TestEntity.java [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+  private OffsetDateTime offsetDateTimeField;
+  private Boolean booleanField;
+  private UUID uuidField;
+
+  /**
+   * Constructs new instance of {@link TestEntity}.
+   */
+  public TestEntity() {
+    byteField = getRandomByte();
+    ;
```
Comment (andrey-yurzanov) [inline entity2jooq-example/entity2jooq-example-h2/src/main/java/io/github/devastool/entity2jooq/example/TestEntity.java:73]: unnecessary `;`

File: entity2jooq-example/entity2jooq-example-h2/src/test/java/io/github/devastool/entity2jooq/example/TestEntityTest.java [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+      new TestEntity(),
+      new TestEntity()
+  );
+
+  @BeforeAll
+  static void init() throws SQLException {
+    pool = JdbcConnectionPool.create(DB_URL, "", "");
+
+    Connection connection = pool.getConnection();
+    DSLContext context = DSL.using(connection);
```
Comment (andrey-yurzanov) [inline entity2jooq-example/entity2jooq-example-h2/src/test/java/io/github/devastool/entity2jooq/example/TestEntityTest.java:71]: Use dialect, please
`DSLContext context = DSL.using(connection, SqlDialects.H2);`

**Review by andrey-yurzanov — CHANGES_REQUESTED** [review]
Some comments

**Review by andrey-yurzanov — APPROVED** [review]

**Review by Sergio-Konovalov — APPROVED** [review]

**Pull request merged**